src/relay/pass/to_gnf.cc

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright (c) 2018 by Contributors
+ *
+ * \file to_gnf.cc
+ *
+ * \brief Turn A normal form into graph normal form.
+ */
+#include <tvm/relay/pass.h>
+#include <tvm/relay/expr_functor.h>
+#include "let_list.h"
+
+namespace tvm {
+namespace relay {
+
+class GNF : public ExprMutator {
+  std::unordered_map<Var, Expr, NodeHash, NodeEqual> var_map_;
+  Expr VisitExpr_(const VarNode* vn) override {
+    Var v = GetRef<Var>(vn);
+    return var_map_.count(v) == 0 ? v : var_map_.at(v);
+  }
+
+  static bool UseVar(const Var& v, const Expr& e) {
```

Comment:
Let us make a quick util function for UseVar, since FreeVars is not the best way to do so